rust/src/delta.rs

```diff
@@ -627,8 +626,45 @@ impl DeltaTable {
             .get_tombstones()
             .iter()
             .filter(|tombstone| tombstone.deletionTimestamp < delete_before_timestamp)
+            .collect())
+    }
+
+    /// Run the Vacuum command on the Delta Table: delete files no longer referenced by a Delta table and are older than the retention threshold.
+    /// We do not recommend that you set a retention interval shorter than 7 days, because old snapshots and uncommitted files can still be in use by concurrent readers or writers to the table. If vacuum cleans up active files, concurrent readers can fail or, worse, tables can be corrupted when vacuum deletes files that have not yet been committed.
+    pub async fn vacuum(
+        &mut self,
+        retention_hours: u64,
+        dry_run: bool,
+    ) -> Result<Vec<String>, DeltaTableError> {
+        let tombstones = self.gc(retention_hours)?;
+        let tombstones_path = tombstones
+            .iter()
             .map(|tombstone| self.storage.join_path(&self.table_path, &tombstone.path))
-            .collect::<Vec<String>>())
+            .collect::<Vec<String>>();
+
+        if dry_run {
+            return Ok(tombstones_path);
+        }
+
+        let mut deleted_tombstones = vec![];
+        for tombstone in tombstones {
+            match self
+                .storage
+                .delete_obj(&self.storage.join_path(&self.table_path, &tombstone.path))
+                .await
+            {
+                Ok(_) => deleted_tombstones.push(tombstone.clone()),
+                Err(StorageError::NotFound) => deleted_tombstones.push(tombstone.clone()),
+                Err(_) => continue,
+            }
+        }
+
+        self.state
+            .tombstones
+            .retain(|tombstone| !deleted_tombstones.contains(tombstone));
+        //TODO: Commit the new state
```

Comment:
@houqp `update` the table is the right method to use for updating the DeltaTable state?

Reply:
not QP level authority - but yes - this is what we use to update table state after committing a new tx as well.

Reply:
**Updated** - now you are 😄